### ISSUE TITLE
Set model to eval mode for knn and linear classification

### ISSUE THREAD

**Issue by guarin** [lightly-ai/lightly#1440]
Our knn and linear classifiers used for benchmarking do not set the model to evaluation mode before training. Instead we only set requires gradients to False, see: https://github.com/lightly-ai/lightly/blob/54a1b7bced3e0b34b1f4c47c13f5eff2347bcbb4/lightly/utils/benchmarking/knn_classifier.py#L144-L150

The correct way to do this would be to set the model to eval mode before every training epoch:
```
def on_train_epoch_start(self) -> None:
    if self.freeze_model:
        self.model.eval()
```

Note that it has to happen in `on_train_epoch_start` because pytorch lightning automatically sets models to train mode at the beginning of the training epoch.

This was already implemented in https://github.com/lightly-ai/lightly/pull/1263 but the PR is not yet merged to the main branch (and probably won't be merged for a while).
